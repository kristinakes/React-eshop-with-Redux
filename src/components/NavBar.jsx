import { useState} from "react";
import { useSelector } from "react-redux";
import { NavLink } from "react-router-dom";
import { Dialog } from "@headlessui/react";
import { Bars3Icon, XMarkIcon } from "@heroicons/react/24/outline";

import laptop from "../assets/icons/laptop.png";

function NavBar() {
  //Redux store data for cart total items
  const cartTotalQiantity = useSelector((state) => state.cart.totalQuantity);

  const navLinks = [
    { name: "Home", href: "/" },
    { name: "Products", href: "/Products" },
    { name: "Contact us", href: "/Contacts" },
    { name: `Cart (${cartTotalQiantity})`, href: "/Cart" },
  ];
  const [mobileMenuOpen, setMobileMenuOpen] = useState(false);

  return (
    <header className="sticky top-0">
      <nav className="bg-blue flex h-[90px]">
        <div className="flex justify-between items-center w-full pl-10 pr-5">
          <NavLink
            to="/"
            className="flex items-center justify-center text-white font-[600] text-2xl gap-3 "
          >
            <img src={laptop} alt="" />
            Gatget Store
          </NavLink>
          <div className="flex lg:hidden">
            <button
              type="button"
              className="-m-2.5 inline-flex items-center justify-center rounded-md p-2.5 text-white"
              onClick={() => setMobileMenuOpen(true)}
            >
              <Bars3Icon className="h-6 w-6" aria-hidden="true" />
            </button>
          </div>
          <div className="hidden lg:flex space-x-6">
            {navLinks.map((link) => {
              return (
                <NavLink
                  key={link.name}
                  to={link.href}
                  className="text-white hover:text-gray-300 text-xl font-600"
                >
                  {link.name}
                </NavLink>
              );
            })}
          </div>
        </div>
        <Dialog
          as="div"
          className="lg:hidden"
          open={mobileMenuOpen}
          onClose={setMobileMenuOpen}
        >
          {/* <div className="fixed inset-0 z-50" /> */}
          <Dialog.Panel className="fixed inset-y-0 right-0 z-50 w-full overflow-y-auto bg-white px-6 py-6 sm:max-w-sm sm:ring-1 sm:ring-gray-900/10">
            <div className="flex items-center justify-between">
              <button
                type="button"
                className="-m-2.5 rounded-md p-2.5 text-gray-700"
                onClick={() => setMobileMenuOpen(false)}
              >
                <span className="sr-only">Close menu</span>
                <XMarkIcon className="h-6 w-6" aria-hidden="true" />
              </button>
            </div>
            <div className="mt-6 flow-root">
              <div className="-my-6 divide-y divide-gray-500/10">
                <div className="space-y-2 py-6">
                  {navLinks.map((link) => (
                    <NavLink
                      key={link.name}
                      to={link.href}
                      className="-mx-3 block rounded-lg px-3 py-2 text-base font-semibold leading-7 text-gray-900 hover:bg-gray-50 active"
                    >
                      {link.name}
                    </NavLink>
                  ))}
                </div>
              </div>
            </div>
          </Dialog.Panel>
        </Dialog>
      </nav>
    </header>
  );
}
export default NavBar;
