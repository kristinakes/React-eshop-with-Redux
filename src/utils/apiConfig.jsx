export const API_PRODUCTS_URL = "http://localhost:8000/products/";
export const API_CART_URL = "http://localhost:8000/cart/";