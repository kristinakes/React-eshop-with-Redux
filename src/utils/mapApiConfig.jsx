//Google maps API key to be used in contacts page. Key should be protected. Update accordingly.
///More info https://developers.google.com/maps/documentation/javascript/get-api-key
export const GOOGLE_API_KEY = "Add your API Key here";