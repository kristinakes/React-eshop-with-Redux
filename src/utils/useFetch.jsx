import { useState, useEffect } from "react";

function useFetch(url, options) {
  const [data, setData] = useState(null);
  const [isPending, setIsPending] = useState(true);
  const [error, setError] = useState(null);

  useEffect(()=> {
    setTimeout(()=> {
        fetch(url, options).then(result => {
            if (!result.ok) {
                throw Error('Error fetching data');
            }
            return result.json();
        }).then(data => {
            setData(data);
            setIsPending(false);
            setError(null);
        }).catch(err => {
            setIsPending(false);
            setError(err.message)
        });
    }, 1000);
  }, [url, options]);

  return {data, isPending, error};
}
export default useFetch;
