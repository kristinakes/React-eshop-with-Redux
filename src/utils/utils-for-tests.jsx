import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import cartSlice from "../store/cartSlice";
import uiSlice from "../store/uiSlice";
import productSlice from "../store/productSlice"

export function renderWithProviders(
  ui,
  {
    preloadedState = {},

    store = configureStore({
      reducer: {
        products: productSlice.reducer,
        cart: cartSlice.reducer,
        ui: uiSlice.reducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
