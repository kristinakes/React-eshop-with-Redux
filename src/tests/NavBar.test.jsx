import { MemoryRouter } from "react-router-dom";

import NavBar from "../components/NavBar";
import { renderWithProviders } from "../utils/utils-for-tests";

describe('Navigation Bar', () => {
    test('should match the snapshot', () => {
        const view = renderWithProviders(
          <MemoryRouter>
            <NavBar />
          </MemoryRouter>
        );

        expect(view.baseElement).toMatchSnapshot();
    })
})