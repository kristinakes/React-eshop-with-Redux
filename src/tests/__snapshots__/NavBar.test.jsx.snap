// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar should match the snapshot 1`] = `
<body>
  <div>
    <header
      class="sticky top-0"
    >
      <nav
        class="bg-blue flex h-[90px]"
      >
        <div
          class="flex justify-between items-center w-full pl-10 pr-5"
        >
          <a
            aria-current="page"
            class="flex items-center justify-center text-white font-[600] text-2xl gap-3  active"
            href="/"
          >
            <img
              alt=""
              src="laptop.png"
            />
            Gatget Store
          </a>
          <div
            class="flex lg:hidden"
          >
            <button
              class="-m-2.5 inline-flex items-center justify-center rounded-md p-2.5 text-white"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
          <div
            class="hidden lg:flex space-x-6"
          >
            <a
              aria-current="page"
              class="text-white hover:text-gray-300 text-xl font-600 active"
              href="/"
            >
              Home
            </a>
            <a
              class="text-white hover:text-gray-300 text-xl font-600"
              href="/Products"
            >
              Products
            </a>
            <a
              class="text-white hover:text-gray-300 text-xl font-600"
              href="/Contacts"
            >
              Contact us
            </a>
            <a
              class="text-white hover:text-gray-300 text-xl font-600"
              href="/Cart"
            >
              Cart (0)
            </a>
          </div>
        </div>
        <div
          style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
        />
      </nav>
    </header>
  </div>
  <div
    id="headlessui-portal-root"
  >
    <div
      data-headlessui-portal=""
    >
      <div />
    </div>
  </div>
</body>
`;
